archetype/engine-v1/src/test/java/io/helidon/build/archetype/engine/v1/PrompterTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.archetype.engine.v1;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PrompterTest {
+
+    @Test
+    void testPromptWithDefaultAnswer() {
+        String input = "\n";
+        System.setIn(new ByteArrayInputStream(input.getBytes()));
+        String result = Prompter.prompt("Enter your name", "defaultName");
+        assertEquals("defaultName", result);
```

Comment:
Please use hamcrest in all tests, only use JUnit assertions as a special case (E.g. `assertThrows`).
I.e.
```java
assertThat(result, is("defaultName"));
```

Reply:
Okay, sure